From 6fc82b275cba3c459a77f6ba71102c5c004b9296 Mon Sep 17 00:00:00 2001
From: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
Date: Fri, 3 Jul 2026 11:07:21 +0100
Subject: [PATCH 09310/12543] irqchip/renesas-rzg2l: Use
 devm_reset_control_get_exclusive_deasserted()

[ Upstream commit 78f384dad082af13a9399b14ff23c5ea02b0a407 ]

Use devm_reset_control_get_exclusive_deasserted() to simplify
rzg2l_irqc_common_init().

Signed-off-by: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Reviewed-by: Philipp Zabel <p.zabel@pengutronix.de>
Reviewed-by: Lad Prabhakar <prabhakar.mahadev-lad.rj@bp.renesas.com>
Link: https://lore.kernel.org/all/20250212182034.366167-3-fabrizio.castro.jz@renesas.com
Signed-off-by: Biju Das <biju.das.jz@bp.renesas.com>
Reviewed-by: Pavel Machek <pavel@nabladev.com>
Signed-off-by: Nobuhiro Iwamatsu <nobuhiro.iwamatsu.x90@mail.toshiba>
---
 drivers/irqchip/irq-renesas-rzg2l.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/irqchip/irq-renesas-rzg2l.c b/drivers/irqchip/irq-renesas-rzg2l.c
index 53272237d9cb..7cf3d6963935 100644
--- a/drivers/irqchip/irq-renesas-rzg2l.c
+++ b/drivers/irqchip/irq-renesas-rzg2l.c
@@ -562,14 +562,10 @@ static int rzg2l_irqc_common_init(struct device_node *node, struct device_node *
 		return ret;
 	}
 
-	resetn = devm_reset_control_get_exclusive(dev, NULL);
-	if (IS_ERR(resetn))
+	resetn = devm_reset_control_get_exclusive_deasserted(dev, NULL);
+	if (IS_ERR(resetn)) {
+		dev_err(dev, "failed to acquire deasserted reset: %d\n", ret);
 		return PTR_ERR(resetn);
-
-	ret = reset_control_deassert(resetn);
-	if (ret) {
-		dev_err(dev, "failed to deassert resetn pin, %d\n", ret);
-		return ret;
 	}
 
 	pm_runtime_enable(dev);
@@ -609,7 +605,6 @@ static int rzg2l_irqc_common_init(struct device_node *node, struct device_node *
 	pm_runtime_put(dev);
 pm_disable:
 	pm_runtime_disable(dev);
-	reset_control_assert(resetn);
 	return ret;
 }
 
-- 
2.53.0

